learning-bitmark/working-with-bitmarks/payment-for-bitmark-cli.md

```diff
@@ -0,0 +1,261 @@
+
+* A correct amount
+
+* To the correct address
+
+* With the correct payment id added as bitcoin/litecoin OP_RETURN opcode.
+
+Because common wallets do not support attaching the Bitmark blockchain payment identifier to the ltc/btc transactions, we provide an app called the Bitmark Wallet to pay for the Bitmark transactions come from the Bitmark CLI.
+
+<br>
+<br>
+
+## Bitmark Wallet
+
+The Bitmark Wallet is a dual currency wallet, supporting both bitcoin and litecoin. It allows sending multiple payments to different addresses and attaching a hex data item to a transaction using the OP_RETURN feature of bitcoin/litecoin.
+
+The wallet is written in Go and cannot work standalone, it can only send the bitcoin and litecoin transactions via a correct configured bitcoind and litecoind. Depending on the configuration of bitcoind and litecoind, the payment transactions are submitted to the mainnet coins or testnet coins.
+
+Supported commands:
+
+* Submit btc/ltc transactions.
+
+* Update the wallet from the btc/ltc block chain - for this it will query its local daemon to request any transctions belonging to its addresses.
+
+* Display the wallet balance for each coin.
+
+* Generate and display an address that can be used to send more coins to the wallet.
+
+> **NOTE:**The Bitmark Wallet uses a master bip39 based recovery phrase which must be printed and kept safe.
+
+<br>
+
+Next steps:
+
+* Install the Bitmark Wallet.
+
+* Sync the wallet with the coins.
+
+* Run the wallet.
+
+* Get the testnet coins.
+
+<br>
+<br>
+
+### Prerequisites of installing the Bitmark Wallet
+
+* Git has been installed.
+
+* Go has been installed.
+
+* Python3 is installed.
+
+* bitcoind/litecoind have been installed.
+
+<br>
+<br>
+
+### Installing the Bitmark Wallet
+
+* Clone the Bitmark Wallet
+
+    ```shell
+        $git clone https://github.com/bitmark-inc/bitmark-wallet.git
+    ```
+
+<br>
+
+* Install the Bitmark Wallet
+
+    ```shell
+        $cd bitmark-wallet/command/bitmark-wallet
+
+        $go install
+    ```
+
+<br>
+
+* Create Bitmark Wallet's config file
+
+    ```shell
+        $ touch <CONFIG DIR>/bitmark-wallet/bitmark-wallet.conf
+    ```
+
+    *Example:*
+    ```shell
+        $ mkdir ${HOME}/.config/bitmark-wallet
+        $ touch ${HOME}/.config/bitmark-wallet/bitmark-wallet.conf
+    ```
+
+<br>
+
+* Copy the content from the [sample config file](samples/bitmark-wallet.conf) to the created config file above
+
+<br>
+<br>
+
+### Connect the Bitmark Wallet with bitcoind and litecoind
+
+* Download the [rpcauth.py](samples/rpcauth.py)
+
+<br>
+
+* Get the ltc and btc rpcauth, rpc username, and rpc password by running
+
+    ```shell
+        $ python3 rpcauth.py localhost
+    ```
+
```

Comment:
When you get something, I always like to show the return, so that a user can then know they're doing it right.

It'd be particularly helpful here, since you're getting info that's then used in the next step. So, readers could see the info output here, then see it used in the next step. It gives them a nice throughline.

Reply:
Updated.